test/WebJobs.Script.Tests/Configuration/ExtensionBundleOptionsSetupTest.cs

```diff
@@ -0,0 +1,184 @@
+            ExtensionBundleOptions options = new ExtensionBundleOptions();
+            var ex = Assert.Throws<ArgumentException>(() => setup.Configure(options));
+            Assert.StartsWith($"The id and version property are missing in extensionBundle section of {ScriptConstants.HostMetadataFileName} file", ex.Message);
+        }
+
+        [Theory]
+        [InlineData(@"{
+                    'version': '2.0',
+                    'extensionBundle': {
+                        'id': null,
+                        'version': '1.0.0'
+                        }
+                    }")]
+        [InlineData(@"{
+                    'version': '2.0',
+                    'extensionBundle': {
+                        'id': 'Microsoft,,',
+                        'version': '1.0.0'
+                        }
+                    }")]
+        [InlineData(@"{
+                    'version': '2.0',
+                    'extensionBundle': {
+                        'version': '1.0.0'
+                        }
+                    }")]
+        public void ValidateBundleId_InvalidId_Throws(string hostJsonContent)
+        {
+            File.WriteAllText(_hostJsonFile, hostJsonContent);
+            Assert.True(File.Exists(_hostJsonFile));
+            var configuration = BuildHostJsonConfiguration();
+
+            ExtensionBundleOptionsSetup setup = new ExtensionBundleOptionsSetup(configuration);
+            ExtensionBundleOptions options = new ExtensionBundleOptions();
+            var ex = Assert.Throws<ArgumentException>(() => setup.Configure(options));
+            Assert.StartsWith($"The value of id property in extensionBundle section of {ScriptConstants.HostMetadataFileName} file is invalid or missing.", ex.Message);
+        }
+
+        [Theory]
+        [InlineData(@"{
+                    'version': '2.0',
+                    'extensionBundle': {
+                        'id': 'Microsoft.Azure.Functions.ExtensionBundle',
+                        'version': null
+                        }
+                    }")]
+        [InlineData(@"{
+                    'version': '2.0',
+                    'extensionBundle': {
+                        'id': 'Microsoft.Azure.Functions.ExtensionBundle',
+                        'version': '(1.0.0'
+                        }
+                    }")]
+        [InlineData(@"{
+                    'version': '2.0',
+                    'extensionBundle': {
+                        'id': 'Microsoft.Azure.Functions.ExtensionBundle',
+                        'version': 'random'
+                        }
+                    }")]
+        [InlineData(@"{
+                    'version': '2.0',
+                    'extensionBundle': {
+                        'id': 'Microsoft.Azure.Functions.ExtensionBundle'
+                        }
+                    }")]
+        public void ExtensionBundleConfigure_InvalidVersion_ThrowsException(string hostJsonContent)
```

Comment:
Use member data instead of InlineData

Reply:
Keeping this in place since the data with the tests would make it easier to debug the tests. And the test data is not re-used anywhere.